engine/runtime-compiler/src/main/scala/org/enso/compiler/context/LocalScope.scala

```diff
@@ -34,11 +35,27 @@ class LocalScope(
   final val aliasingGraph: () => AliasGraph,
   final private val scopeProvider: () => AliasGraph.Scope,
   final private val dataflowInfoProvider: () => DataflowAnalysis.Metadata,
-  final val flattenToParent: Boolean                       = false,
-  private val parentFrameSlotIdxs: Map[AliasGraph.Id, Int] = Map()
+  final private val symbolsProvider: () => FramePointerAnalysis.FramePointerMeta =
+    null,
+  final val flattenToParent: Boolean                             = false,
+  private val parentFrameSlotIdxs: () => Map[AliasGraph.Id, Int] = () => Map()
 ) {
   lazy val scope: AliasGraph.Scope                 = scopeProvider()
   lazy val dataflowInfo: DataflowAnalysis.Metadata = dataflowInfoProvider()
+  def allSymbols(where: String): List[String] = {
+    def symbols() = scope.allDefinitions.map(_.symbol)
+    val meta      = if (symbolsProvider == null) null else symbolsProvider()
+    if (meta != null && meta.variableNames.isDefined) {
+      val cached = meta.variableNames.get
+      FramePointer.assertSame(where, cached, () => symbols())
```

Comment:
Remove the `assert` or not? 
- If I keep it, then running with `-ea` is going to be slow as it always computes symbols from `AliasGraph`
- If I remove it, then we risk an inconsistency somewhere - but all our know code is currently processed OK...

Reply:
Done in c87f67a0b65c926cf1b3c880150f4576b7a91475 - use:
```bash
JAVA_OPTS=-ea ./enso --log-level trace --run xyz.enso
```
e.g. _enable assertions_ and also logging to get this check.

Reply:
Can you put this to test somehow? Create a context with trace log level, pipe the log to some `ByteArrayBuffer` and ensure there are no assertion errors? That way, this will be not checked in every test, but it will at least be tested **somewhere**.